sycl/include/sycl/detail/generic_type_traits.hpp

```diff
@@ -28,51 +27,66 @@ namespace sycl {
 inline namespace _V1 {
 namespace detail {
 template <typename T>
-inline constexpr bool is_svgenfloatf_v =
-    is_contained_v<T, gtl::scalar_vector_float_list>;
+using is_byte = typename
+#if (!defined(_HAS_STD_BYTE) || _HAS_STD_BYTE != 0)
+    std::is_same<T, std::byte>;
+#else
+    std::false_type;
+#endif
+
+template <typename T> inline constexpr bool is_byte_v = is_byte<T>::value;
 
 template <typename T>
-inline constexpr bool is_svgenfloath_v =
-    is_contained_v<T, gtl::scalar_vector_half_list>;
+inline constexpr bool is_svgenfloatf_v =
+    std::is_same_v<T, float> ||
+    (is_vec_v<T> && std::is_same_v<element_type_t<T>, float>);
 
 template <typename T>
-inline constexpr bool is_genfloat_v = is_contained_v<T, gtl::floating_list>;
+inline constexpr bool is_svgenfloath_v =
+    std::is_same_v<T, half> ||
+    (is_vec_v<T> && std::is_same_v<element_type_t<T>, half>);
 
 template <typename T>
 inline constexpr bool is_sgenfloat_v =
-    is_contained_v<T, gtl::scalar_floating_list>;
+    check_type_in_v<T, float, double, half, ext::oneapi::bfloat16>;
```

Comment:
given that you replace `is_contained_v` with `check_type_in_v` its implementation should probably be changed to the `is_base_of` way of checking membership for best performance

Reply:
Yeah, I'll look into that separately.